android/libraries/rib-base/src/main/java/com/badoo/ribs/core/routing/configuration/feature/operation/BackStackOperation.kt

```diff
@@ -0,0 +1,9 @@
+package com.badoo.ribs.core.routing.configuration.feature.operation
+
+import android.os.Parcelable
+import com.badoo.ribs.core.routing.configuration.feature.BackStackElement
+
+interface BackStackOperation<C : Parcelable> {
+    fun isApplicable(backStack: List<BackStackElement<C>>): Boolean
+    fun modifyStack(backStack: List<BackStackElement<C>>): List<BackStackElement<C>>
```

Comment:
1. typealias BackStack<C> = List<BackStackElement<C>>
2. functional interface here + invoke() in call sites?

```suggestion
interface BackStackOperation<C : Parcelable> : (BackStack<C>) -> BackStack<C> {
    fun isApplicable(backStack: BackStack<C>): Boolean
```